omnigan/trainer.py

```diff
@@ -1560,216 +1558,84 @@ def run_evaluation(self, verbose=0):
-
-    def resume(self, inference=False):
-        tpu = "xla" in str(self.device)
-        if tpu:
-            print("Resuming on TPU:", self.device)
-
-        m_path = Path(self.opts.load_paths.m)
-        p_path = Path(self.opts.load_paths.p)
-        pm_path = Path(self.opts.load_paths.pm)
-        output_path = Path(self.opts.output_path)
-
-        map_loc = self.device if not tpu else "cpu"
-
-        if "m" in self.opts.tasks and "p" in self.opts.tasks:
-            # ----------------------------------------
-            # -----  Masker and Painter Loading  -----
-            # ----------------------------------------
-
-            # want to resume a pm model but no path was provided:
-            # resume a single pm model from output_path
-            if all([str(p) == "none" for p in [m_path, p_path, pm_path]]):
-                checkpoint_path = output_path / "checkpoints/latest_ckpt.pth"
-                print("Resuming P+M model from", str(checkpoint_path))
-                checkpoint = torch.load(checkpoint_path, map_location=map_loc)
-
-            # want to resume a pm model with a pm_path provided:
-            # resume a single pm model from load_paths.pm
-            # depending on whether a dir or a file is specified
-            elif str(pm_path) != "none":
-                assert pm_path.exists()
-
-                if pm_path.is_dir():
-                    checkpoint_path = pm_path / "checkpoints/latest_ckpt.pth"
-                else:
-                    assert pm_path.suffix == ".pth"
-                    checkpoint_path = pm_path
-
-                print("Resuming P+M model from", str(checkpoint_path))
-                checkpoint = torch.load(checkpoint_path, map_location=map_loc)
-
-            # want to resume a pm model, pm_path not provided:
-            # m_path and p_path must be provided as dirs or pth files
-            elif m_path != p_path:
-                assert m_path.exists()
-                assert p_path.exists()
-
-                if m_path.is_dir():
-                    m_path = m_path / "checkpoints/latest_ckpt.pth"
-
-                if p_path.is_dir():
-                    p_path = p_path / "checkpoints/latest_ckpt.pth"
-
-                assert m_path.suffix == ".pth"
-                assert p_path.suffix == ".pth"
-
-                m_checkpoint = torch.load(m_path, map_location=map_loc)
-                p_checkpoint = torch.load(p_path, map_location=map_loc)
-                checkpoint = merge(m_checkpoint, p_checkpoint)
-                print(f"Resuming P+M model from \n  -{p_path} \nand \n  -{m_path}")
+    def eval_images(self, mode, domain):
+        if domain == "rf":
+            return
 
-            else:
-                raise ValueError(
-                    "Cannot resume a P+M model with provided load_paths:\n{}".format(
-                        self.opts.load_paths
-                    )
-                )
+        metrics = {"accuracy": accuracy, "mIOU": mIOU}
```

Comment:
I think you are not setting num_classes in the mIOU (it matters because in our case it's 11 and not the default 19 from the mIOU function ? )

Reply:
right, fixed